src/main/java/connectify/logic/commands/ShareCompanyCommand.java

```diff
@@ -0,0 +1,121 @@
+package connectify.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.List;
+
+import connectify.commons.core.index.Index;
+import connectify.commons.util.ToStringBuilder;
+import connectify.logic.Messages;
+import connectify.logic.commands.exceptions.CommandException;
+import connectify.model.Model;
+import connectify.model.company.Company;
+
+/**
+ * Shares a company to another address book.
+ */
+public class ShareCompanyCommand extends Command {
+
+    public static final String COMMAND_WORD = "shareCompany";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD
+            + ": Shares instructions on how to add a Company to another address book.\n"
+            + "Parameters: "
+            + "INDEX (must be a positive integer)\n"
+            + "Example: " + COMMAND_WORD + " 1";
+
+    public static final String MESSAGE_SHARE_COMPANY_SUCCESS = "Command to add this Company:\n%1$s\n"
+            + "Do take note that you need to add people on your own.";
+
+    private final Index targetIndex;
+
+    /**
+     * Creates a ShareCompanyCommand to share the specified {@code Company}
+     */
+    public ShareCompanyCommand(Index targetIndex) {
+        requireNonNull(targetIndex);
+        this.targetIndex = targetIndex;
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
```

Comment:
Good idea implementing this method in this manner, which models the way existing commands are executed!